src/main/java/duke/TaskCommand.java

```diff
@@ -0,0 +1,124 @@
+package duke;
+
+import java.io.IOException;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+/**
+ * Runs the command for user to create a new task.
+ */
+
+public class TaskCommand extends Command {
+
+    enum TaskType {
+        TODO, DEADLINE, EVENT
+    }
+
+    private TaskType taskType;
+    private String command;
+
+    /**
+     * Constructor to create a command to create a new task.
+     * @param taskType determine what kind of task to be created.
+     * @param command the command entered by the user.
+     */
+    TaskCommand (TaskType taskType, String command) {
+        this.taskType = taskType;
+        this.command = command;
+    }
+
+    /**
+     * Executes the command proceeds to create a new task for the user, and appending it to the list.
+     *
+     * @param taskList The TaskList of the current user.
+     * @param ui The user interface to show messages to users.
+     * @param storage The file system for reading and writing into the database.
+     * @throws IOException when there is an error opening the file.
+     * @throws ParseException when there is an error in recognising the text.
+     */
+    @Override
+    String runCommand(TaskList taskList, Ui ui, Storage storage) throws IOException {
```

Comment:
Hello! I am allocated to view your code in tutorial. I think this method is too long. Also, it is a bit deep nesting and forms an arrow-head style. You can consider abstracting out the lower-level methods, such as the `Parse()` function, because they are not in the same level as those higher-level methods like `addTask()`. That will make the code look more cleaner :)